### PR TITLE
feat: delay hydration / fix flickering for signals

### PR DESCRIPTION
**Pull request by dunqan**
Closes [HRZ-2618](https://spryker.atlassian.net/browse/HRZ-2618)

---

### Checklist before requesting a code review

- [x] PR title is aligned with [PR titles and conventional commits](https://spryker.atlassian.net/wiki/spaces/FA/pages/3821175291/PR+titles+and+conventional+commits)
- [x] I have performed a self-review of my code and have fixed all jobs failures

[HRZ-2618]: https://spryker.atlassian.net/browse/HRZ-2618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/base/utilities/src/signals/signal-from.ts [hunk 1]
```diff
@@ -11,8 +11,35 @@ export interface ConnectableSignalOptions<T, K> extends SignalOptions<T | K> {
...
+    if (!_resolvingSignals) return false;
+    if (!_resolvingSignalsNotifier)
+      _resolvingSignalsNotifier = new StateSignal<number>(_resolvingSignals!);
+    return _resolvingSignalsNotifier.value;
+  };
+}
+
 export class SignalObservable<T, K = undefined> extends StateSignal<T | K> {
   subscription?: Subscription;
+  protected resolving = true;
```
Comment (tolerants) [inline libs/base/utilities/src/signals/signal-from.ts:46]: isResolving?

File: libs/base/utilities/src/signals/signal-from.ts [hunk 2]
```diff
@@ -11,8 +11,35 @@ export interface ConnectableSignalOptions<T, K> extends SignalOptions<T | K> {
   initialValue?: K;
 }
 
+let _resolvingSignals: number | undefined = undefined;
+let _resolvingSignalsNotifier: StateSignal<number> | undefined = undefined;
+
+function signalResolving(state = true): void {
```
Comment (tolerants) [inline libs/base/utilities/src/signals/signal-from.ts]: Having two similar names for functions (`signalResolving` and `resolvingSignals`) can be confusing. Suggest to rename and maybe write a small description.

Comment (dunqan) [reply libs/base/utilities/src/signals/signal-from.ts]: 👍

**Review by supproduction — APPROVED** [review]

File: libs/base/ui/action/button/src/button.component.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import { IconTypes } from '@spryker-oryx/ui/icon';
-import { Size } from '@spryker-oryx/utilities';
+import { hydratable, Size } from '@spryker-oryx/utilities';
 import { html, LitElement, TemplateResult } from 'lit';
 import { property } from 'lit/decorators.js';
 import { ButtonComponentAttributes, ButtonType } from './button.model';
 import { buttonStyles } from './styles';
 
+@hydratable()
```
Comment (tobi-or-not-tobi) [inline libs/base/ui/action/button/src/button.component.ts:8]: why is this one hydrated?

**Review by tobi-or-not-tobi — APPROVED** [review]

File: libs/base/ui/action/button/src/button.component.ts (hunk 3 above)
Comment (dunqan) [reply libs/base/ui/action/button/src/button.component.ts:8]: it has loading logic, so at least has to be hydratable, when attributes would change

**Pull request merged**